relay-server/src/services/processor/span/processing.rs

```diff
@@ -91,10 +98,18 @@ pub fn process(
             return ItemAction::Drop(Outcome::Invalid(DiscardReason::Internal));
         };
 
+        if should_extract_transactions && !item.transaction_extracted() {
+            if let Some(transaction) = convert_to_transaction(&annotated_span) {
```

Comment:
Span and transaction normalization may have different requirements, so I'd extract transactions before normalizing spans (line 91 above).